From f5d90753363caae81ea681f6d3152420fc94cb01 Mon Sep 17 00:00:00 2001
From: Ben Bright <ben.bright@twdb.texas.gov>
Date: Fri, 17 Feb 2023 18:54:27 +0000
Subject: [PATCH] Changes for patch
 0005-Added-check-for-content-type-header-in-proxy_resource on 2.9.8

---
 ckanext/resourceproxy/blueprint.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/ckanext/resourceproxy/blueprint.py b/ckanext/resourceproxy/blueprint.py
index 8ea9efa73b..65501b581c 100644
--- a/ckanext/resourceproxy/blueprint.py
+++ b/ckanext/resourceproxy/blueprint.py
@@ -68,7 +68,15 @@ def proxy_resource(context, data_dict):
         if not did_get:
             r = requests.get(url, timeout=TIMEOUT, stream=True)
 
-        response.headers[u'content-type'] = r.headers[u'content-type']
+        if 'content-type' in r.headers:
+            response.headers[u'content-type'] = r.headers[u'content-type']
+        else:
+            return abort(
+                409, (
+                u'"content-type" header not found.'
+                )
+            )
+
         response.charset = r.encoding
 
         length = 0
-- 
2.25.1

